src/ocean_emulators/train.py

```diff
@@ -139,6 +134,20 @@ def __init__(self, cfg: TrainConfig) -> None:
         self.N_bound = len(self.boundary_var_names)
         self.N_prog = len(self.prognostic_var_names)
 
+        self.data_container = cfg.data.build(
+            data_root=cfg.experiment.resolved_data_root,
+            boundary_var_names=self.boundary_var_names,
+        )
+
+        self.mp_context: BaseContext | None
+        if cfg.data.num_workers > 0:
+            if self.data_container.supports_fork:
+                self.mp_context = multiprocessing.get_context("fork")
+            else:
+                self.mp_context = multiprocessing.get_context("spawn")
+        else:
+            self.mp_context = None
```

Comment:
🐑  I prefer this style, as it has one less branch, but I see the merits of what's already here, too.

```suggestion
        self.mp_context: BaseContext | None = None
        if cfg.data.num_workers > 0:
            if self.data_container.supports_fork:
                self.mp_context = multiprocessing.get_context("fork")
            else:
                self.mp_context = multiprocessing.get_context("spawn")
```